### ISSUE TITLE
Error on clang-format plugin

### ISSUE THREAD

**Issue by nelson777** [siduck76/NvChad#2024]
I'm trying to install NvChad, but I'm getting this error: 

```
 Failed
     󰚌 clang-format
      ▶ # [13/13] Failed to create python3 virtual environment.

```

**To Reproduce**
Just run the installation 

**Expected behavior**
Installation finishing correctly and some kind of initial screen appearing.

**Screenshots**
![Captura de tela de 2023-05-15 06-54-55](https://github.com/NvChad/NvChad/assets/5481589/bd56c840-5ee4-4910-9b00-60a96e20d762)

**Desktop (please complete the following information):**
 -Linux Mint 21 Vanessa
 - Terminator/Bash
 - 0.9.0

**Additional context**
I'm using Nvim 0.9 build from source, after many problems trying to install it. If you guys think it's relevant I can write my whole ordeal trying to install it. But it's long and I guess it's solved. At least the health checks pass.

I want to use the NVim's stable version.

**Comment by siduck** [comment]
enable python3 provider https://github.com/NvChad/NvChad/issues/1417

**Comment by nelson777** [comment]
Well what I did was downloading NvChat enviroment and before running, I created a init.vim file with this inside it:

```
let g:loaded_python3_provider = 1 
let  g:python3_host_prog="/usr/bin/python3" 

```

The config files should have conflicted, but to my surprise it worked and the installation finished. But when I tried to load nvim after installation I got a config conflict error. So I deleted the init.vim file.
But now the python provider doesn't load:

![Captura de tela de 2023-05-15 08-04-49](https://github.com/NvChad/NvChad/assets/5481589/c589fd3d-9d6d-4953-9549-ca52799ba15d)

I tryed to go to lua/core/init.lua and include:

vim.g.loaded_python3_provider = 1

but it didn't work also. I also tried removing python3 from this part:

```
-- disable some default providers
for _, provider in ipairs { "node", "python3", "perl", "ruby" } do
  vim.g["loaded_" .. provider .. "_provider"] = 0
end

```

But it didn't work also. Any ideas ?

**Comment by siduck** [comment]
```lua
local enable_providers = {
  "python3_provider",
  "node_provider",
  -- and so on
}

for _, plugin in pairs(enable_providers) do
  vim.g["loaded_" .. plugin] = nil
  vim.cmd("runtime " .. plugin)
end
```

**Comment by nelson777** [comment]
Eh... where's that file ? sorry I never user NVim... I'm starting. Is this init.vim  on ./config/nvim ?

**Comment by siduck** [comment]
> Eh... where's that file ? sorry I never user NVim... I'm starting. Is this init.vim on ./config/nvim ?

custom/init.lua . Please read our docs from start to end

**Comment by nelson777** [comment]
Sorry, I took a better look at the docs. But I don't have a custom dir in ~/.config/nvim/lua 
I tried to put that file in that path but I got the error:

```
Error executing vim.schedule lua callback: /home/nelson/.config/nvim/lua/core/utils.lua:80: attempt to index a nil value
stack traceback:
        /home/nelson/.config/nvim/lua/core/utils.lua:80: in function </home/nelson/.config/nvim/lua/core/utils.lua:55>

```
What I did:

As I'm on Linux Mint 21 Vanessa, I uninstalled completely with: 

```
rm -rf ~/.config/nvim
rm -rf ~/.local/share/nvim
```

Then, to be able to change the installation before the start I did: 

```
git clone https://github.com/NvChad/NvChad ~/.config/nvim --depth 1
```

Then I checked ~/.config/nvim/lua and it didn't contain a custom dir. So I created it and put the lines you mentioned in the init.lua file.

Then I started the installation with nvim command. It started ok but then I got the error above.

What I'm I doing wrong now ?

**Comment by siduck** [comment]
```
rm -rf ~/.config/nvim
rm -rf ~/.local/share/nvim
```
and now install nvchad, it will prompt you to select example_config and press Y

**Comment by nelson777** [comment]
Well I don't think how I can explain better what the problem is. But let me try again:

The problem is that clang-format plugin installtion fails and makes the installation stall. I do have selected to create example_config at installation.

your advice for this is that I put 

```
local enable_providers = {
  "python3_provider",
  "node_provider",
  -- and so on
}

for _, plugin in pairs(enable_providers) do
  vim.g["loaded_" .. plugin] = nil
  vim.cmd("runtime " .. plugin)
end
```
in config.lua right ? but to put that there I have to have an working installation. 
I tried to just download install before running removing `&& nvim` from install command in order to put that code in the installation before starting it but it didn't work because there are many more files needed beyond config.lua.

So how can I put that code there BEFORE INSTALLATION starts, so I can run installation with python provider active and make installation fininsh correctly ?

Also please reopen issue as the problem wasn't solved.

**Comment by siduck** [comment]
hi @nelson777 sorry my bad for not directing you well! just read the issue properly now and its not related to nvchad, read this mason issue https://github.com/williamboman/mason.nvim/issues/729

**Comment by nelson777** [comment]
Ah... now it worked. Thanks. :) 
I understand that we are all with lots of things in our backs. No problem. :)
